### ISSUE TITLE
Move "track" stats to the obsolete section

### ISSUE THREAD

**Issue by henbos** [w3c/webrtc-stats#479]
After moving everything significant from `track`/`sender`/`receiver` to `outbound-rtp` and `inbound-rtp` there is little use for track attachment stats, or even sender/receiver stats.

```
track/sender
  // From RTC[Video/Video]SenderStats:
  mediaSourceId
  // No members are defined in RTC[Audio/Video]HandlerStats.
  // From RTCMediaHandlerStats:
  trackIdentifier
  remoteSource
  ended
  kind
  priority

track/receiver
  // No members are defined in RTC[Audio/Video][Receiver/Handler]Stats.
  // From RTCMediaHandlerStats:
  trackIdentifier
  remoteSource
  ended
  kind
  priority
```

"mediaSourceId", "trackIdentifier" and "kind" already exists on `outbound-rtp` and `inbound-rtp`, these are not needed.

Sender-side: "ended" should probably move to `media-source` since this is a stat about the local track.
Receiver-side: "ended" should perhaps move to `inbound-rtp`, though in that case it would be more elegant to define it to reflect that the RTP stream has terminated, it should be equivalent.

"priority" is a feature at risk and probably doesn't hurt to be removed, but if we want to keep it then perhaps it should be added to the parameters stats proposed in #398.

"remoteSource" is only needed to be able to tell a sending `track` from a receiving `track` and is not needed for `sender` or `receiver` or `outbound-rtp` or `inbound-rtp`, i.e. this is not an argument for keeping track attachment stats.

**Comment by henbos** [comment]
TL;DR action items unless an argument is made for keeping them:
- Move all the track/sender/receiver related dictionaries (the whole hierarchy) to the obsolete section. For specific members, this means...
- Add "ended" to `media-source`, `inbound-rtp` and the obsolete section.
- Add "trackIdentifier" and "kind" to the obsolete section, these already exists in the *-rtp dictionaries and don't need moving.
- Add "remoteSource" to the obsolete section, it is no longer needed.
- Add "priority" to the obsolete section; if #398 is resolved also move it to the encoding stats, otherwise let it not have a replacement.

**Comment by henbos** [comment]
Another use case of track attachment stats are not the stats themselves but the fact that they trigger `onstatsended`, see #480 for what to do about that...

**Comment by henbos** [comment]
One more thing, in the case of simulcast, there is still a point to be made about having sender - or transceiver (#478) - stats if they get a "mid" (#396) so that you can correlate which RTP streams belong to which sender/transceiver.

**Comment by henbos** [comment]
TPAC decision: Remove "track" stats

**Comment by henbos** [comment]
"track" stats are to be made obsolete. "sender" and "receiver" stats remain for the sake of "transceiver" stats and simulcast use cases.

**Comment by henbos** [comment]
I cut this up into smaller action items. I think now that "sender" and "receiver" stats are not going away we are free to make "track" obsolete separately from moving other things to obsolete section or moving other metrics around.

- Move RTCSenderVideoTrackAttachmentStats and RTCSenderAudioTrackAttachmentStats to obsolete section (the receiving "track" stats are already in the obsolete section). Filed #527.
- "remoteSource": Move to obsolete section. Filed #528.
- "priority" is obsolete. Filed #529.
- "ended" can remain where it is because we're keeping "sender" and "receiver" stats. The same is true for "trackIdentifier" and "kind". However we may want to revisit if we want to move some of these to "media-source" and if the receiving track-related stats should be on the "receiver" or on the "inbound-rtp". Filed #530
